app/WebParsing/UtsgJsonParser.hs

```diff
@@ -42,17 +44,54 @@ insertAllMeetings org = do
     manager <- liftIO $ newManager tlsManagerSettings
     response <- liftIO $ httpLbs request' manager
     let respBody = responseBody response
+    return respBody
 
-    -- decode the response
+-- Get the page number, page size and total number of courses from response
+getPageInfo :: ByteString -> Maybe (Int, Int, Int)
+getPageInfo respBody = do
+    jsonValue <- decode respBody
+    case jsonValue of 
+      Nothing -> return (-1, -1, -1)
+      Just json -> do 
+        flip parseMaybe json $ \obj -> do
+          payload :: Object <- obj .: "payload"
+          pageableCourse :: Object <- payload .: "pageableCourse"
+          page :: Int <- pageableCourse .: "page"
```

Comment:
I don't think you need type annotations here, as Haskell should be able to infer the expected type of `page`, `pageSize`, and `totalCourses` from the function signature